core/templates/dev/head/services/assets-backend-api.service.spec.ts

```diff
@@ -59,39 +66,71 @@ describe('Assets Backend API Service', function() {
       'AssetsBackendApiService');
     fileDownloadRequestObjectFactory = $injector.get(
       'FileDownloadRequestObjectFactory');
+    audioFileObjectFactory = $injector.get('AudioFileObjectFactory');
+    imageFileObjectFactory = $injector.get('ImageFileObjectFactory');
     UrlInterpolationService = $injector.get(
       'UrlInterpolationService');
     $httpBackend = $injector.get('$httpBackend');
     $rootScope = $injector.get('$rootScope');
     ENTITY_TYPE = $injector.get('ENTITY_TYPE');
     $q = $injector.get('$q');
+
+    CsrfService = $injector.get('CsrfTokenService');
+
+    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
+      var deferred = $q.defer();
+      deferred.resolve('sample-csrf-token');
+      return deferred.promise;
+    });
+
+    audioRequestUrl = UrlInterpolationService.interpolateUrl(
+      '/assetsdevhandler/exploration/<exploration_id>/assets/audio/<filename>',
+      {
+        exploration_id: '0',
+        filename: 'myfile.mp3'
+      });
+
+    imageRequestUrl = UrlInterpolationService.interpolateUrl(
+      '/assetsdevhandler/exploration/<exploration_id>/assets/image/<filename>',
+      {
+        exploration_id: '0',
+        filename: 'myfile.png'
+      });
+
+    window.BlobBuilder = undefined;
   }));
 
   afterEach(function() {
     $httpBackend.verifyNoOutstandingRequest();
   });
 
-  it('should correctly formulate the download URL', function() {
-    // TODO(sll): Find a way to substitute out constants.DEV_MODE so that we
-    // can test the production URL, too.
```

Comment:
is this issue resolved?

Reply:
Yes. It was resolved in the last `describe` block.